pkg/kernelsyms/kernel_symbols.go

```diff
@@ -0,0 +1,86 @@
+package kernelsyms
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+type KernelSymbolTable struct {
+	symbolMap map[string]KernelSymbol
+}
+
+type KernelSymbol struct {
+	Name    string
+	Type    string
+	Address uint64
+	Owner   string
+}
+
+type HookedSyscallData struct {
+	SyscallName string
+	ModuleOwner string
+}
+
+const IoctlSyscallHook int = 65
+
+/*
+ * this package is to support a kernel-malware detection and especially rootkits.
+ * in addition its gives tracee the ability to holdall the known kernel symbols
+ *
+ * the KernelSymbolTable type holds map of all the kernel symbols with a ke y which is the kernel object owner and the name with undercase between them
+ * which means that symbolMap looks like [objectOwner_objectname{SymbolData}, objectOwner_objectname{SymbolData}, etc...]
+ * the key naming is because sometimes kernel symbols can have the same name or the same address which prevents to key the map with only one of them
+ *
+ */
+
+//GetSymbolByAddr return's a symbol by a given address
+func (t *KernelSymbolTable) GetSymbolByAddr(addr uint64) (KernelSymbol, error) {
+	for _, Symbol := range t.symbolMap {
+		if Symbol.Address == addr {
+			return Symbol, nil
+		}
+	}
+	return KernelSymbol{}, fmt.Errorf("couldent find the symbol")
+}
```

Comment:
Now its called  for every hooked syscall so I think it isn't supposed to to called so frequently because we check less then 20 syscalls and also I don't think any rootkit will hook more then 5 syscalls.
(for example diamorphine which is in if the most popular rootkits in GitHub hooks only 3 syscalls)